### PR TITLE
Feature: add logistic regression and multiple regression as options

### PR DESCRIPTION
**Pull request by martinctc**
# Summary
This branch adds logistic regression and multiple regression as additional options. This version is incremented to `0.1.2.9000` as the development version (not submitted to CRAN). 

The changes made in this PR are:

1. **Core functions**: Created `rwa_logit()` for binary outcomes (logistic regression) and `rwa_multiregress()` for continuous outcomes (multiple regression)
2. **Wrapper function**: Converted `rwa()` to automatically detect method based on outcome variable (2 unique values → logistic, otherwise → multiple regression) with explicit `method` parameter override
3. **API standardization**: Unified output structure across both methods (consistent column naming, rescaled weights sum to 100, lambda matrices)
4. **Numerical stability**: Added safeguards in `rwa_logit()` for edge cases (clipping predictions away from 0/1, zero standard deviation handling)
5. **Documentation**: Added comprehensive roxygen2 docs with examples for both functions and new "Regression Methods" vignette
6. **Plotting support**: Extended `plot_rwa()` to handle both regression types with appropriate labeling
7. **Test coverage**: Added test suites for logistic regression, integration tests, and validation of numerical edge cases

**Usage example:**
```r
# Automatic detection - binary outcome uses logistic regression
mtcars$high_mpg <- ifelse(mtcars$mpg > median(mtcars$mpg), 1, 0)
rwa(mtcars, "high_mpg", c("cyl", "disp", "hp", "wt"))

# Automatic detection - continuous outcome uses multiple regression  
rwa(mtcars, "mpg", c("cyl", "disp", "hp", "wt"))

# Explicit method selection
rwa(mtcars, "high_mpg", c("cyl", "disp", "hp", "wt"), method = "logistic")
```

***

# Check
- [x] All R CMD checks pass 
- [x] `roxygen2::roxygenise()` has been run prior to merging to ensure that `.Rd` and `NAMESPACE` files are up to date.

### CODE REVIEW

File: R/rwa.R [hunk 1]
```diff
@@ -54,180 +79,158 @@
...
-  matrix_data <-
-    cor_matrix %>%
-    as.matrix()
+  outcome_values <- unique(df[[outcome]])
+  outcome_var_unique <- dplyr::n_distinct(outcome_values[!is.na(outcome_values)])
 
-  RXX <- matrix_data[2:ncol(matrix_data), 2:ncol(matrix_data)] # Only take the correlations with the predictor variables
-  RXY <- matrix_data[2:ncol(matrix_data), 1] # Take the correlations of each of the predictors with the outcome variable
+  use_logistic <- (method == "logistic") ||
+                  (method == "auto" && outcome_var_unique == 2)
```
[the remark this answers is not in the document]
Comment (martinctc) [reply R/rwa.R:180]: The `method` parameter already exists specifically to override auto-detection - users can explicitly set `method = "multiple"` or `method = "logistic"`.

**Comment by martinctc** [comment]
@copilot Can you help summarise the changes in this PR in a numbered bulleted list?

**Comment by martinctc** [comment]
All checks passing - merging `0.1.2.9000` into main now.
